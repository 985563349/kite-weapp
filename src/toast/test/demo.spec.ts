import path from 'node:path';
import simulate from 'miniprogram-simulate';

test('should render demo and match snapshot', async () => {
  const comp = simulate.render(
    simulate.load(path.resolve(__dirname, '../demo/toast'), {
      rootPath: path.resolve(__dirname, '../../'),
    })
  );
  comp.attach(document.createElement('parent-wrapper'));

  expect(comp.toJSON()).toMatchSnapshot();
});
