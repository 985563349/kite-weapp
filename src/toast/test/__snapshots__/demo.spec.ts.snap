// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render demo and match snapshot 1`] = `
<main>
  <k-flex
    gap="md"
    style="padding-bottom: 16px"
    vertical="{{true}}"
  >
    <demo-block
      title="基础用法"
    >
      <k-space
        wrap="{{true}}"
      >
        <k-button
          type="primary"
          bind:click="showTextToast"
        >
          Text
        </k-button>
        <k-button
          type="primary"
          bind:click="showLoadingToast"
        >
          Loading
        </k-button>
        <k-button
          type="primary"
          bind:click="showSuccessToast"
        >
          Success
        </k-button>
        <k-button
          type="primary"
          bind:click="showWarningToast"
        >
          Warning
        </k-button>
        <k-button
          type="primary"
          bind:click="showErrorToast"
        >
          Error
        </k-button>
      </k-space>
    </demo-block>
    <demo-block
      title="弹出位置"
    >
      <k-space>
        <k-button
          type="primary"
          bind:click="showTopToast"
        >
          Top
        </k-button>
        <k-button
          type="primary"
          bind:click="showBottomToast"
        >
          Bottom
        </k-button>
      </k-space>
    </demo-block>
  </k-flex>
  <k-toast
    id="k-toast"
  />
</main>
`;
